cmake/CMakeLists.txt

```diff
@@ -0,0 +1,178 @@
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
+    set(dialect "-fdefault-real-8 -fdefault-double-8 -ffree-line-length-none -fbacktrace")
+    set(bounds "-fbounds-check")
+    if(NOT CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER_EQUAL 8.0.0)
+      set(DISABLE_OMP ON)
+    endif()
+endif()
+if(CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
+    set(dialect "-axAVX2 -r8 -traceback")
+    set(bounds "-check bounds")
+endif()
+set(CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS_DEBUG} ${bounds}")
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${dialect}")
+
+# Populate xtb_version.fh
+set(version ${PROJECT_VERSION})
+execute_process(COMMAND git rev-parse HEAD
+ RESULT_VARIABLE git_return
+ OUTPUT_VARIABLE commit)
+string(REGEX REPLACE "\n$" "" commit ${commit})
+string(TIMESTAMP date "%m/%d/%Y")
+set(author $ENV{USERNAME})
+set(origin $ENV{HOSTNAME})
+configure_file("${XTB_ROOT}/xtb/version.f90" "${XTB_ROOT}/xtb/xtb_version.fh" @ONLY)
+
+# Find dependencies
+if (NOT DISABLE_OMP)
+  find_package(OpenMP REQUIRED)
+endif()
+find_package(LAPACK REQUIRED)
+find_package(BLAS REQUIRED)
+
+# Object library
+add_library(xtb_object OBJECT ${XTB_SOURCES})
+target_link_libraries(xtb_object
+ PUBLIC
+  $<$<BOOL:${OpenMP_Fortran_FOUND}>:OpenMP::OpenMP_Fortran>
+)
+set_target_properties(xtb_object PROPERTIES
+  Fortran_MODULE_DIRECTORY xtb-mod
+  POSITION_INDEPENDENT_CODE ON
+)
+target_include_directories(xtb_object
+ PUBLIC
+  $<BUILD_INTERFACE:${XTB_ROOT}/include>
+  $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include/xtb>
+)
+
+# Static Library
+add_library(xtb_static STATIC $<TARGET_OBJECTS:xtb_object> )
+target_link_libraries(xtb_static
+ PUBLIC
+  ${BLAS_LIBRARIES}
+  ${LAPACK_LIBRARIES}
+  $<$<BOOL:${OpenMP_Fortran_FOUND}>:OpenMP::OpenMP_Fortran>
+)
+set_target_properties(xtb_static PROPERTIES
+  Fortran_MODULE_DIRECTORY xtb-mod
+  ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
+  POSITION_INDEPENDENT_CODE ON
+)
+target_include_directories(xtb_static
+ PUBLIC
+  $<BUILD_INTERFACE:${XTB_ROOT}/include>
+  $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include/xtb>
+)
+
+# Shared Library
+add_library(xtb_shared SHARED $<TARGET_OBJECTS:xtb_object> )
+target_link_libraries(xtb_shared
+ PUBLIC
+  ${BLAS_LIBRARIES}
+  ${LAPACK_LIBRARIES}
+  $<$<BOOL:${OpenMP_Fortran_FOUND}>:OpenMP::OpenMP_Fortran>
+)
+set_target_properties(xtb_shared PROPERTIES
+  Fortran_MODULE_DIRECTORY xtb-mod
+  LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
+)
+target_include_directories(xtb_shared
+ PUBLIC
+  $<BUILD_INTERFACE:${XTB_ROOT}/include>
+  $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include/xtb>
+)
+
+# Executables
+add_executable(xtb "${XTB_ROOT}/xtb/program_main.f90")
+target_link_libraries(xtb
+ PRIVATE
+  xtb_static
+)
+set_target_properties(xtb PROPERTIES
+  Fortran_MODULE_DIRECTORY xtb-mod
+  RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
+)
+target_include_directories(xtb PRIVATE "${XTB_ROOT}/include")
+
+# Install
+install(FILES "${XTB_ROOT}/include/xtb.h"
+        DESTINATION include/xtb/
```

Comment:
Sure it should be `include/xtb`?

Reply:
Well I am used to more than just one header file for a program, so that is the default.

Reply:
meson puts it by default to `include`, but I want consistency, so we should change the meson build instructions. Could you apply this patch?
```patch
diff --git a/meson.build b/meson.build
index 96c1d39..eb185c1 100644
--- a/meson.build
+++ b/meson.build
@@ -20,7 +20,8 @@ project('xtb',
         'fortran', 'c',
         version: '6.2.2',
         license: 'LGPL3',
-        meson_version: '>=0.49')
+        meson_version: '>=0.49',
+        default_options: ['includedir=include/xtb'])
 
 fc = meson.get_compiler('fortran')
 cc = meson.get_compiler('c')
```

Reply:
Done